subset.py

```diff
@@ -125,30 +125,16 @@ def __subset_upstream(self, target, processed_tables, relationships):
         if len(relevant_key_constraints) == 0 or target in processed_tables:
             return False
 
-        temp_target_name = 'subset_temp_' + table_name(target)
-
-        try:
-            # copy the whole table
-            columns_query = columns_to_copy(target, relationships, self.__source_conn)
-            self.__db_helper.run_query('CREATE TEMPORARY TABLE {} AS SELECT * FROM {} LIMIT 0'.format(quoter(temp_target_name), fully_qualified_table(mysql_db_name_hack(target, self.__destination_conn))), self.__destination_conn)
-            query = 'SELECT {} FROM {}'.format(columns_query, fully_qualified_table(target))
-            self.__db_helper.copy_rows(self.__source_conn, self.__destination_conn, query, temp_target_name)
-
-            # filter it down in the target database
-            table_columns = self.__db_helper.get_table_columns(table_name(target), schema_name(target), self.__source_conn)
-            clauses = ['{} IN (SELECT {} FROM {})'.format(columns_tupled(kc['fk_columns']), columns_joined(kc['target_columns']), fully_qualified_table(mysql_db_name_hack(kc['target_table'], self.__destination_conn))) for kc in relevant_key_constraints]
-            clauses.extend(upstream_filter_match(target, table_columns))
-
-            select_query = 'SELECT * FROM {} WHERE TRUE AND {}'.format(quoter(temp_target_name), ' AND '.join(clauses))
-            insert_query = 'INSERT INTO {} {}'.format(fully_qualified_table(mysql_db_name_hack(target, self.__destination_conn)), select_query)
-            self.__db_helper.run_query(insert_query, self.__destination_conn)
-            self.__destination_conn.commit()
-
-        finally:
-            # delete temporary table
-            mysql_temporary = 'TEMPORARY' if config_reader.get_db_type() == 'mysql' else ''
-            self.__db_helper.run_query('DROP {} TABLE IF EXISTS {}'.format(mysql_temporary, quoter(temp_target_name)), self.__destination_conn)
-
+        columns_query = columns_to_copy(target, relationships, self.__source_conn)
+        table_columns = self.__db_helper.get_table_columns(table_name(target), schema_name(target), self.__source_conn)
+        filters = upstream_filter_match(target, table_columns)
+        clauses = ['{} IN (SELECT {} FROM {})'.format(columns_tupled(kc['fk_columns']), columns_joined(kc['target_columns']), fully_qualified_table(mysql_db_name_hack(kc['target_table'], self.__source_conn))) for kc in relevant_key_constraints]
+        query = 'SELECT {} FROM {} WHERE TRUE AND {}'.format(columns_query, fully_qualified_table(target), ' AND '.join(clauses))
+        if len(filters) > 0:
+            query += ' AND ' + ' AND '.join(filters) 
+        query += " LIMIT {}".format(config_reader.get_max_rows_per_table())
+        self.__db_helper.copy_rows(self.__source_conn, self.__destination_conn, query, table_name(target))
+        
```

Comment:
Yargh, unfortunately this doesn't work. See these queries actually run against subqueries of other tables in order to determine which rows to pull, and this needs to be run on the destination, otherwise it will pull against the full tables in the source rather than the already subset tables in the destination for determining which rows to pull in the current upstream table.

In order to do this in a better way, we'd need to cache the foreign key values during processing, which we don't do in condenser's implementation. Otherwise, this leads to a subset that doesn't make sense, unfortunately.

Reply:
Yea, this makes sense. I started to get a bit overzealous here. 

In my specific use case, a lot of the data I want to filter is time-series data which I was limiting with upstream filters. so the upstream filters are a bigger limit than the referential subsetting. It was taking a very long time to make the copy of the source tables with quite a lot of data that would just be filtered out by the upstream filters. I will roll back these changes from this pr, but I do wonder if it would be possible to apply the upstream filters earlier?